challenges.yml

```diff
@@ -2088,3 +2088,217 @@ extending_prop_transition:
+        b: |
+          When `v-model.capitalize` is used, Vue capitalizes every word entered
+          in an input element.
+        c: |
+          When `v-model.numner` is used, Vue parses the value into a JavaScript
+          `Number` type.
+      correct: b
+
+extend_v_model_custom_component:
+  question: |
+    We in the process of building a nice checkout form for our pizza customers.
+    It looks something like this:
+
+    ```vue
+    <template>
+        <form>
+            <p>Please, enter your info here:</p>
+
+            <fancy-input
+                id="name"
+                v-model="client.name"
+                label="Name:"
+            />
+        </form>
+    </template>
+    ```
+
+    We have just written the first input for the customer's name and we want it
+    to be a custom component but we've forgotten how to implement `v-model` in
+    a custom component in Vue 2!
+
+    Which of these templates would do the trick, assuming our component has a prop
+    `value` of type `String`?
+  explanation: |
+    In vue 2, `v-model` translates to `:value` and an `$emit` with the event
+    `input` and the text to be changed! Just remember that in Vue 3, the
+    default prop to be used in `v-model` is `modelValue`!
+
+    > Alternatively, you can also pass `v-model:value` in Vue 3. The `:value` part
+    > acts as a parameter of the `v-model` *"function"* and tells it that
+    > the prop we want to use is called `value`. Just like in Vue 2!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```vue
+          <input
+              :id="id"
+              :name="id"
+              type="text"
+              :value="value"
+              @input="$emit('input', $event.target.value)"
+          >
+          ```
+        b: |
+          ```vue
+          <input
+              :id="id"
+              :name="id"
+              type="text"
+              :value="value"
+              @input="$emit('input', $event)"
+          >
+          ```
+        c: |
+          ```vue
+          <input
+              :id="id"
+              :name="id"
+              type="text"
+              :value="value"
+              @change="$emit('input', $event.target.value)"
+          >
+          ```
+        d: |
+          ```vue
+          <input
+              :id="id"
+              :name="id"
+              type="text"
+              :value="value"
+              @keyup="$emit('input', $event.target.value)"
+          >
+          ```
+      correct: a
+extend_v_bind_object:
+  question: |
+    `v-bind` is an amazing shortcut that lets us assign multiple, or even dynamic
+    props to a component using one single attribute. Given this piece of code,
+    what are the actual props that the child component will receive?
+
+    ```vue
+    <template>
+        <child-component
+            text="Some text prop here"
+            v-bind="{
+                isValid: true,
+                showText: true,
+            }"
```

Comment:
We could make it even more interesting by "overriding" a value in `v-bind`, like:

```
            text="Some text prop here"
            :isValid="false"
            v-bind="{
                isValid: true,
                showText: true,
            }"
```

Then below, we can talk about the override logic. I think it's just as simple as, in this example, `v-bind` comes second, and so the "later" props override the first. If you moved `v-bind` to the top then the `:isValid="false"` would win (we should double-check this ;) )

Reply:
There's nothing in the docs that talk about this. Have you seen this in action? If not, I will have to do some tests (we also don't show this in the course)

Reply:
Ok, discussed, tested, and decided to leave as is.